### PR TITLE
MED-217: Customer Status Labels & Internal Notes

### PR DESCRIPTION
**Pull request by qinshi599**
Implements MED-217 — customer status labels and internal notes (full stack, project-scoped).

**Status labels:** admin create/rename/recolor/reorder/delete; in-use deletion shows an in-app confirm modal; duplicate name in a project rejected inline; assignable on the customer record and shown as a badge on the conversation profile panel.

**Internal notes:** rich-text notes on the customer profile, never visible to the customer; show author name/avatar + time; author edits/deletes own, admin deletes any; all deletions audit-logged (NFR-011).

**QA feedback addressed:** native pop-ups replaced with in-app ConfirmModal; edge cases added (demo + automated tests); fixed a duplicate-name 500 → clean 400.

Tests: backend 83 passing, frontend ColorPicker + StatusLabelModal.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: backend/customer/models.py [hunk 1]
```diff
@@ -107,4 +115,143 @@ class Meta:
...
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        ordering = ['order', 'name']
+        constraints = [
+            models.UniqueConstraint(
+                fields=['project', 'name'],
+                name='customer_status_label_unique_name_per_project',
+            )
```
[the remark this answers is not in the document]
Comment (qinshi599) [reply backend/customer/models.py:147]: this is a real but low-risk edge case. The serializer already enforces case-insensitive uniqueness (name__iexact), which covers normal usage. The remaining gap is a narrow concurrency race (two simultaneous creates of "Gold"/"gold"), and the worst case is two similar labels — no crash or data leak. Given labels are configured infrequently by admins, I'll leave the case-insensitive DB constraint as a follow-up rather than ship a migration for it in this ticket.

**Review by quanwangniuniu — APPROVED** [review]
Good job!
